packages/client/components/PinnedSnackbarNotifications.tsx

```diff
@@ -0,0 +1,109 @@
+import graphql from 'babel-plugin-relay/macro'
+import {useFragment} from 'react-relay'
+import {PinnedSnackbarNotifications_query$key} from '~/__generated__/PinnedSnackbarNotifications_query.graphql'
+import useAtmosphere from '../hooks/useAtmosphere'
+import useRouter from '../hooks/useRouter'
+import SetNotificationStatusMutation from '../mutations/SetNotificationStatusMutation'
+import mapTeamsLimitExceededToToast from '../mutations/toasts/mapTeamsLimitExceededToToast'
+import {OnNextHistoryContext} from '../types/relayMutations'
+import {NotificationEnum} from '../__generated__/popNotificationToast_notification.graphql'
+import {Snack} from './Snackbar'
+
+interface Props {
+  queryRef: PinnedSnackbarNotifications_query$key
+}
+
+const typePicker: Partial<
+  Record<NotificationEnum, (notification: any, context: OnNextHistoryContext) => Snack | null>
+> = {
+  TEAMS_LIMIT_EXCEEDED: mapTeamsLimitExceededToToast
+}
+
+const PinnedSnackbarNotifications = ({queryRef}: Props) => {
+  const data = useFragment(
+    graphql`
+      fragment PinnedSnackbarNotifications_query on Query {
+        viewer {
+          pinnedNotifications: notifications(first: 10, types: [TEAMS_LIMIT_EXCEEDED]) {
+            edges {
+              node {
+                id
+                status
+                type
+                ...mapTeamsLimitExceededToToast_notification @relay(mask: false)
+              }
+            }
+          }
+        }
+      }
+    `,
+    queryRef
+  )
+  const {history} = useRouter()
+  const atmosphere = useAtmosphere()
+  const {viewer} = data
+  const notifications = viewer?.pinnedNotifications || {edges: []}
+  const {edges} = notifications
+  const snackbarNotifications = edges.filter(
+    ({node}) => node.status === 'UNREAD' && Object.keys(typePicker).includes(node.type)
+  )
+
+  snackbarNotifications.forEach(({node}) => {
+    const specificNotificationToastMapper = typePicker[node.type]
+    if (!specificNotificationToastMapper) {
+      return
+    }
+
+    const notificationSnack = specificNotificationToastMapper(node, {
+      atmosphere,
+      history
+    })
+
+    if (!notificationSnack) {
+      return
+    }
+
+    if (notificationSnack.autoDismiss === 0) {
+      const callback = notificationSnack.onDismiss
+      notificationSnack.onDismiss = () => {
+        const {id: notificationId} = node
+        SetNotificationStatusMutation(
+          atmosphere,
+          {
+            notificationId,
+            status: 'CLICKED'
```

Comment:
-1 I don't understand why this is necessary. It's the same event as calling the action. In fact I see both events are fired here.

Reply:
@Dschoordsch snackbar is attached to a specific notification. We want to mark notification clicked, if snackbar is clicked. However, snackbar could also be dismissed automatically, in this case we don't want `clicked` mutation to be executed.

Refactored code a bit, and added `onManualDismiss` that should cover necessary cases.

Reply:
@Dschoordsch

>  but there is some precedence in the code where we set CLICKED just to set the notification inactive

That is exactly what I'm trying to do here too.

If the user manually dismiss a snackbar that is attached to a specific notification, I just mark corresponding notification as `clicked`, so it will not appear as unread and will not be highlighted in notifications dropdown.